### DISCUSSION TITLE
Support for Sentinel NAT Mapping

### DISCUSSION THREAD

**Discussion by daveteu** [redis/go-redis#2130] [Ideas]
Is there any support for NAT mapping?

For example, our Sentinels and redis servers are connected via LAN IP addresses (faster), and hence, the sentinel announces the LAN IP address.

However, our API servers may be located **outside** the LAN, which means we need to connect to WAN IP.

Is there a way we can map WAN IP to LAN IP?

E.g. when I was using javascript, we use ioredis with this feature https://github.com/luin/ioredis#nat-mapping

**Comment by vmihailenco** [comment]
You could try to override the dialer and do the mapping there.

**Reply by daveteu** [reply]
Thank you @vmihailenco . I found the example below, and I understand I can do my own mapping using switch or conditions base on address, but with regards to the `Dial` function, do I change `sshClient.Dial(network, addr)` to `net.Dial(network, addr)` ?

And should I remove the `Addr`, `ReadTimeout` and `WriteTimeout` lines?

Thank you in advance, i am really new to this.

```
rdb := redis.NewClient(&redis.Options{
	Addr: net.JoinHostPort("127.0.0.1", "6379"),
	Dialer: func(ctx context.Context, network, addr string) (net.Conn, error) {
		return sshClient.Dial(network, addr)
	},
	// Disable timeouts, because SSH does not support deadlines.
	ReadTimeout:  -1,
	WriteTimeout: -1,
})
```
